Errors when accessing supposed method from a non-returning function's value

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** 
**OS:** 

```
OS: macos, macOS, 11.2.1, 20D75
Processor: 16 cpus, 64bit, little endian, Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
CC version: Apple clang version 12.0.5 (clang-1205.0.22.11)

getwd: /Users/bryanhu
vmodules: /Users/bryanhu/.vmodules
vroot: /Users/bryanhu/.bootstrap/v
vexe: /Users/bryanhu/.bootstrap/v/v
vexe mtime: 2021-06-23 16:24:57
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.2 66bf963

Git version: git version 2.30.1 (Apple Git-130)
Git vroot status: weekly.2021.25-24-g66bf963a (2 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 689c8a02
```

**What did you do?**

Try to *compile* this code:

```v
fn a() {
	println('something')
}
fn main() {
	a().error()
}
```

You pretend `a()` returns a struct with a method `error()`

**What did you expect to see?**

Something along the lines of
 ```
test.v:5:6: cgen error: checker bug; unexpected SelectorExpr.expr_type = 0
    3 | }
    4 | fn main() {
    5 |     a().error()
      |         ~~~~~~
    6 | }
```
Like what would happen if you remove the parenthesis from `.error()` (although that error message is already kind of cryptic. I might want to open a new bug to improve this error message)

**What did you see instead?**
```
==================
                      ^                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/v/test.26863002730894438.tmp.c:10069:2: error: implicit declaration of function 'void_error' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        void_error(main__a());
        ^
/tmp/v/test.26863002730894438.tmp.c:10069:2: note: did you mean 'v_error'?
/tmp/v/test.26863002730894438.tmp.c:7210:15: note: 'v_error' declared here
inline IError v_error(string message) {
              ^
/tmp/v/test.26863002730894438.tmp.c:10069:13: error: argument type 'void' is incomplete
        void_error(main__a());
                   ^~~~~~~~~
1 warning and 2 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

If you were not working with C interop, this is a compiler bug, please report the bug using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -1778,6 +1778,10 @@ pub fn (mut c Checker) method_call(mut call_expr ast.CallExpr) ast.Type {
 	if left_type_sym.kind in [.sum_type, .interface_] && method_name == 'type_name' {
 		return ast.string_type
 	}
+	if left_type == ast.void_type {
+		c.error('`void` type has no methods', call_expr.left.position())
+		return ast.void_type
+	}
 	mut has_generic := false // x.foo<T>() instead of x.foo<int>()
 	mut concrete_types := []ast.Type{}
 	for concrete_type in call_expr.concrete_types {
@@ -2907,6 +2911,7 @@ pub fn (mut c Checker) selector_expr(mut node ast.SelectorExpr) ast.Type {
 		node.name_type = name_type
 		return ast.string_type
 	}
+
 	//
 	old_selector_expr := c.inside_selector_expr
 	c.inside_selector_expr = true
@@ -2915,10 +2920,7 @@ pub fn (mut c Checker) selector_expr(mut node ast.SelectorExpr) ast.Type {
 	//
 	c.using_new_err_struct = using_new_err_struct_save
 	if typ == ast.void_type_idx {
-		// This means that the variable's value was assigned to an
-		// unknown function or method, so the error was already handled
-		// earlier
-		// c.error('unknown selector expression', node.pos)
+		c.error('`void` type has no fields', node.pos)
 		return ast.void_type
 	}
 	node.expr_type = typ

